Original patch from gentoo: ../../dev/crosstool-NG/ct-ng.trunk/patches/strace/4.5.17/130-fix-disabled-largefile-syscalls.patch
-= BEGIN original header =-
diff -dur strace-4.5.16.orig/syscall.c strace-4.5.16/syscall.c
-= END original header =-
diff -durN strace-4.5.17.orig/syscall.c strace-4.5.17/syscall.c
--- strace-4.5.17.orig/syscall.c	2008-10-24 21:30:13.000000000 +0200
+++ strace-4.5.17/syscall.c	2008-10-24 21:30:13.000000000 +0200
@@ -125,6 +125,18 @@
 #define TP TRACE_PROCESS
 #define TS TRACE_SIGNAL
 
+#ifndef HAVE_STATFS64
+/*
+ * Ugly hacks for systems that do not have LFS
+ */
+
+#define sys_truncate64  sys_truncate
+#define sys_ftruncate64 sys_ftruncate
+#define sys_getdents64  sys_getdents
+#define sys_statfs64    sys_statfs
+#define sys_fstatfs64   sys_fstatfs
+#endif
+
 static const struct sysent sysent0[] = {
 #include "syscallent.h"
 };
